lychee-bin/src/main.rs

```diff
@@ -197,16 +198,14 @@ async fn run(cfg: &Config, inputs: Vec<Input>) -> Result<i32> {
     .await
     .map_err(|e| anyhow!(e))?;
 
-    let pb = if cfg.no_progress {
-        None
-    } else {
+    let pb = cfg.no_progress.not().then(|| {
```

Comment:
Does this increase any readability or performance?

Reply:
I think the functional style is clearer but I agree that the `.not()` may not be (see my original comment).

Reply:
The issue is not with `not`, the main issue is the `functional style`.
`std::option::Option::then()` is a fairly new addition. If it's the more idiomatic way of handling Boolean conditionals, it'd be added much earlier.